### PR TITLE
fix kubectl proxy reject-method not use issue

### LINKED ISSUE #6: kubectl proxy --reject-methods flag is non-functional
The flag is registered, but the specified value is never read or used

This means all http methods are allowed by default (despite the help saying PUT/POST/PATCH being disallowed), and users have no way to control which methods are accepted

### PR DESCRIPTION
**Pull request by adohe-zz**
quick fix https://github.com/kubernetes/kubectl/issues/6. @liggitt ptal.

### CODE REVIEW

File: pkg/kubectl/cmd/proxy.go [hunk 1]
```diff
@@ -115,6 +115,7 @@ func RunProxy(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
 		AcceptPaths: kubectl.MakeRegexpArrayOrDie(cmdutil.GetFlagString(cmd, "accept-paths")),
 		RejectPaths: kubectl.MakeRegexpArrayOrDie(cmdutil.GetFlagString(cmd, "reject-paths")),
 		AcceptHosts: kubectl.MakeRegexpArrayOrDie(cmdutil.GetFlagString(cmd, "accept-hosts")),
+		RejectMethods:kubectl.MakeRegexpArrayOrDie(cmdutil.GetFlagString(cmd, "reject-methods")),
```
Comment (liggitt) [inline pkg/kubectl/cmd/proxy.go]: Question is how to release note this and whether to change the default. Will people expect these to be filtered (as documented) or be surprised when things kubectl proxy currently allows stop working by default?

Comment (adohe) [inline pkg/kubectl/cmd/proxy.go]: Nice catch. Personally I think we should add this rejected methods as what we have already documented, this enables a more secure proxy server. It did break the currently things maybe we can add more clear error msg to tell users why you get rejected. wdyt?

Comment (CaoShuFeng) [inline pkg/kubectl/cmd/proxy.go]: I think the default three rejected methods are a little odd.
Why we reject POST,PUT,PATCH? Why not reject DELETE?
I think we can change the default value to "^$" and reject nothing.

Comment (soltysh) [inline pkg/kubectl/cmd/proxy.go]: The last proposal seems to be backwards compatible.

**Comment by suyogbarve** [comment]
@adohe Please close this, issue has been resolved as part of 
https://github.com/kubernetes/kubernetes/commit/4ff9234eec8f35c28397ac70fd167d999e7618ff

**Comment by liggitt** [comment]
closed in https://github.com/kubernetes/kubernetes/pull/44908

**Pull request closed without merging**